XXX-url.md

```diff
@@ -0,0 +1,75 @@
+| Title  | Implement WHATWG URL Spec   |
+|--------|-----------------------------|
+| Author | @jasnell                    |
+| Status | DRAFT                       |
+| Date   | 2016-06-01T09:00:00-07:00   |
+
+## Description
+
+The WhatWG URL Standard specifies updated syntax, parsing and serialization of
+URLs as currently implemented by the main Web Browsers. The existing Node.js
+`url` module parsing and serialization implementation currently does not support
+the URL standard and fails to pass almost half of the parsing tests.
+
+This proposal is to implement the WhatWG URL Standard by introducing a new
+`URL` *global*.
+
+The *existing* `url` module would **remain** is it currently is, with the
+addition of a single `require('url').URL` export that can be used an an
+alternative way of the getting a reference to the new `URL` constructor.
+
+Existing locations within core that currently accept a URL string or the
+parsed results of `require('url').parse()` would be modified to also accept
+the `URL` object. Instances within core that current parse URL strings using
```

Comment:
Ah OK. Then one approach would be type-testing for the existing url module and string coercion for whatever's left, but maybe it's simpler just to do type-testing all the way.